qiita_db/commands.py

```diff
@@ -235,25 +238,26 @@ def load_processed_data_cmd(fps, fp_types, processed_params_table_name,
         raise ValueError("Please pass exactly one fp_type for each "
                          "and every fp")
 
-    fp_types_dict = get_filepath_types()
-    fp_types = [fp_types_dict[x] for x in fp_types]
+    with TRN:
+        fp_types_dict = get_filepath_types()
+        fp_types = [fp_types_dict[x] for x in fp_types]
 
-    if preprocessed_data_id is not None:
-        preprocessed_data = PreprocessedData(preprocessed_data_id)
-    else:
-        preprocessed_data = None
+        if preprocessed_data_id is not None:
```

Comment:
Non blocking but it will be nice IMOO. This block and the next:
preprocessed_data = None if preprocessed_data_id is None else PreprocessedData(preprocessed_data_id)